### PR TITLE
Add flake8-implicit-str-concat check to Ruff

### PR DESCRIPTION
**Pull request by josh-fell**
This check was enabled initially in #23873, but hasn't been part of the Ruff checks to-date. Let's add it!

### CODE REVIEW

**Review by Taragolis — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -188,3 +189,8 @@ exclude_also = [
     "@(typing(_extensions)?\\.)?overload",
     "if (typing(_extensions)?\\.)?TYPE_CHECKING:"
 ]
+
+[tool.vulture]
+min_confidence = 80
+paths = ["airflow/"]
+sort_by_size = true
```
Comment (o-nikolas) [inline pyproject.toml]: I didn't expect to see two changes to this file, what does this one do?

**Comment by o-nikolas** [comment]
Other than one comment, looks good to me! Love this one :heart:

File: pyproject.toml (hunk 1 above)
Comment (josh-fell) [reply pyproject.toml]: Oops, got code bleed from something else I was working on. Thanks for catching!

Comment (josh-fell) [reply pyproject.toml]: Removed now.

**Comment by potiuk** [comment]
> Other than one comment, looks good to me! Love this one ❤️

Likewise :)

**Review by potiuk — APPROVED** [review]

**Review by o-nikolas — APPROVED** [review]

**Review by pankajkoti — APPROVED** [review]
nice, just wondering how those string concatenations existed in so many places :) maybe it was another pre-commit responsible for it.

**Comment by potiuk** [comment]
> nice, just wondering how those string concatenations existed in so many places :) maybe it was another pre-commit responsible for it.

We used to get rid of them by flake I think - but at the time we moved to ruff it was not available as far as I remember. 
And those strings are create by black reformatting - for example when you remove and the whole block gets de-dented previously split strings (because they were too long) are moved to one line. But black (and ruff-black follows it) has the rule that it never EVER touches the AST so concatenating two strings is not possible in ruff.

That's the root cause we had so many of those.

And yes. I got far too many of those as well and it started to be annoying :)

**Review by jscheffl — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**